api/src/test/java/org/openmrs/module/atomfeed/api/service/TagsServiceTest.java

```diff
@@ -0,0 +1,112 @@
+import org.openmrs.module.atomfeed.api.service.impl.TagsServiceImpl;
+
+import javax.xml.bind.JAXBException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.mockito.Mockito.when;
+
+public class TagsServiceTest {
+
+	private static final String FEED_FILTER_XML_PATTERN;
+	private static final String BAD_FEED_FILTER_XML_1;
+	private static final String BAD_FEED_FILTER_XML_2;
+	private static final String BAD_FEED_FILTER_XML_3;
+	private static final int CORRECT_TAGS_NUMBER = 3;
+	private static final String BEAN_NAME_PREFIX = "BeanName-";
+	private static final String FILTER_PREFIX = "Filter-";
+	private static final String CANNOT_PARSE_ERROR = "Cannot parse";
+
+	@InjectMocks
+	private TagsServiceImpl tagsService;
+
+	@Mock
+	private XMLParseService xmlParseService;
+
+	@Before
+	public void setUp() {
+		MockitoAnnotations.initMocks(this);
+	}
+
+	@Test
+	public void testGettingFeedFiltersFromTags() throws JAXBException {
+		List tags = prepareTagsList();
+
+		List<FeedFilter> feedFilters = tagsService.getFeedFiltersFromTags(tags);
+
+		Assert.assertEquals(CORRECT_TAGS_NUMBER, feedFilters.size());
+		for (int i = 0; i < feedFilters.size(); i++) {
+			Assert.assertEquals(BEAN_NAME_PREFIX + i, feedFilters.get(i).getBeanName());
+			Assert.assertEquals(FILTER_PREFIX + i, feedFilters.get(i).getFilter());
+		}
+	}
+
+	private List prepareTagsList() throws JAXBException {
+		List tags = new ArrayList();
+
+		int i = 0;
+		// Add 3 correct Feed Filters
+		for (; i < CORRECT_TAGS_NUMBER; i++) {
+			String feedFilterXML = String.format(FEED_FILTER_XML_PATTERN, BEAN_NAME_PREFIX + i, FILTER_PREFIX + i);
+			when(xmlParseService.createFeedFilterFromXMLString(feedFilterXML))
+					.thenReturn(new FeedFilter(BEAN_NAME_PREFIX + i, FILTER_PREFIX + i));
+			tags.add(feedFilterXML);
+		}
+		// Add 3 bad formatted FeedFilters
+		String feedFilter1 = String.format(BAD_FEED_FILTER_XML_1, BEAN_NAME_PREFIX + i, FILTER_PREFIX + i++);
+		when(xmlParseService.createFeedFilterFromXMLString(feedFilter1))
+				.thenThrow(new JAXBException(CANNOT_PARSE_ERROR));
+		tags.add(feedFilter1);
+		String feedFilter2 = String.format(BAD_FEED_FILTER_XML_2, BEAN_NAME_PREFIX + i, FILTER_PREFIX + i++);
+		when(xmlParseService.createFeedFilterFromXMLString(feedFilter2))
+				.thenThrow(new JAXBException(CANNOT_PARSE_ERROR));
+		tags.add(feedFilter2);
+		String feedFilter3 = String.format(BAD_FEED_FILTER_XML_3, BEAN_NAME_PREFIX + i, FILTER_PREFIX + i);
+		when(xmlParseService.createFeedFilterFromXMLString(feedFilter3))
+				.thenThrow(new JAXBException(CANNOT_PARSE_ERROR));
+		tags.add(feedFilter3);
+		// Add 1 Tag which cannot be parsed to FeedFilter
+		String tag = "patient,CREATED";
+		when(xmlParseService.createFeedFilterFromXMLString(tag))
+				.thenThrow(new JAXBException(CANNOT_PARSE_ERROR));
+		tags.add(tag);
+		// Add 1 object that is not a String
+		tags.add(new Object());
+
+		return tags;
+
+	}
+
+	static {
```

Comment:
you could either use xml files to store those values with proper names, or specify why those XMLs are not valid, as it's not so clear.